sys-utils/rollup/rollup.cpp

```diff
@@ -292,12 +309,13 @@ static int write_advance_state(rollup &r, const cmt_rollup_finish_t *f) {
         {"data",
             {
                 {"chain_id", advance.chain_id},
-                {"payload", hex(reinterpret_cast<const uint8_t *>(advance.payload), advance.payload_length)},
-                {"app_contract", hex(advance.app_contract, sizeof(advance.app_contract))},
-                {"msg_sender", hex(advance.msg_sender, sizeof(advance.msg_sender))},
+                {"app_contract", hex(advance.app_contract.data, sizeof(advance.app_contract))},
```

Comment:
This works, but is kind of brittle. You should probably use `sizeof(advance.app_contract.data)`. Otherwise, if you introduce another field into the structure, this will crash. Better yet, use `std::size(advance.app_contract.data)`, so if you later change data to be a pointer, it will stop compiling altogether and force you to look into this. Same for every time the same `sizeof(struct)` is being used.